registry/remote/repository.go

```diff
@@ -195,6 +195,10 @@ func (r *Repository) FetchReference(ctx context.Context, reference string) (ocis
 // wrapped error ErrInvalidReference.
 func (r *Repository) ParseReference(reference string) (registry.Reference, error) {
 	ref, err := registry.ParseReference(reference)
+	if errors.Is(err, errdef.ErrInvalidReferenceTerminal) {
```

Comment:
Exit immediately on the new terminal error; do not attempt recovery.

Reply:
The error recovery is by design since a `reference` passed to `Repository.ParseReference()` can be a FQDN like `localhost:5000/alpine:latest` or just the tag `latest`.

If we exit immediately, we will miss that later case.